Overall bash-completion expects GNU grep or at least a POSIX-compatible grep.

diff -Nurb bash-completion-2.5.orig/completions/_modules bash-completion-2.5/completions/_modules
--- bash-completion-2.5.orig/completions/_modules	2017-06-02 18:26:34.798318968 +0000
+++ bash-completion-2.5/completions/_modules	2017-06-02 18:28:36.833504279 +0000
@@ -37,7 +37,7 @@
 {
     local modules="$( \
         module avail 2>&1 | \
-        command grep -E -v '^(-|$)' | \
+        command ggrep -E -v '^(-|$)' | \
         xargs printf '%s\n' | command sed -e 's/(default)//g' | sort )"
 
     compgen -W "$modules" -- $1
@@ -53,7 +53,7 @@
         # First parameter on line -- we expect it to be a mode selection
 
         local options
-        options="$( module help 2>&1 | command grep -E '^[[:space:]]*\+' | \
+        options="$( module help 2>&1 | command ggrep -E '^[[:space:]]*\+' | \
                     awk '{print $2}' | command sed -e 's/|/ /g' | sort )"
 
         COMPREPLY=( $(compgen -W "$options" -- "$cur") )
diff -Nurb bash-completion-2.5.orig/completions/adb bash-completion-2.5/completions/adb
--- bash-completion-2.5.orig/completions/adb	2017-06-02 18:26:34.803361858 +0000
+++ bash-completion-2.5/completions/adb	2017-06-02 18:28:36.824112192 +0000
@@ -3,7 +3,7 @@
 _adb_command_usage()
 {
     COMPREPLY=( $( compgen -W \
-        '$( "$1" help 2>&1 | command grep "^ *\(adb \)\? *$2 " \
+        '$( "$1" help 2>&1 | command ggrep "^ *\(adb \)\? *$2 " \
             | command sed -e "s/[]|[]/\n/g" | _parse_help - )' -- "$cur" ) )
 }
 
diff -Nurb bash-completion-2.5.orig/completions/apt-cache bash-completion-2.5/completions/apt-cache
--- bash-completion-2.5.orig/completions/apt-cache	2017-06-02 18:26:34.800291621 +0000
+++ bash-completion-2.5/completions/apt-cache	2017-06-02 18:28:36.830273714 +0000
@@ -7,7 +7,7 @@
 
 # List APT source packages
 _apt_cache_sources() {
-    apt-cache dumpavail | command grep "^Source: $1" | cut -f2 -d" " | sort -u
+    apt-cache dumpavail | command ggrep "^Source: $1" | cut -f2 -d" " | sort -u
 }
 
 # List APT source packages
diff -Nurb bash-completion-2.5.orig/completions/apt-get bash-completion-2.5/completions/apt-get
--- bash-completion-2.5.orig/completions/apt-get	2017-06-02 18:26:34.801560854 +0000
+++ bash-completion-2.5/completions/apt-get	2017-06-02 18:28:36.803075265 +0000
@@ -27,7 +27,7 @@
             source)
                 COMPREPLY=( $( apt-cache --no-generate pkgnames "$cur" \
                     2> /dev/null ) $( apt-cache dumpavail | \
-                    command grep "^Source: $cur" | sort -u | cut -f2 -d" " ) )
+                    command ggrep "^Source: $cur" | sort -u | cut -f2 -d" " ) )
                 ;;
             *)
                 if [[ $special == install && $cur == */* ]]; then
@@ -48,7 +48,7 @@
             ;;
         -t|--target-release|--default-release)
             COMPREPLY=( $( apt-cache policy | \
-                command grep "release.o=Debian,a=$cur" | \
+                command ggrep "release.o=Debian,a=$cur" | \
                 command sed -e "s/.*a=\(\w*\).*/\1/" | uniq 2> /dev/null) )
             return
             ;;
diff -Nurb bash-completion-2.5.orig/completions/aptitude bash-completion-2.5/completions/aptitude
--- bash-completion-2.5.orig/completions/aptitude	2017-06-02 18:26:34.801913603 +0000
+++ bash-completion-2.5/completions/aptitude	2017-06-02 18:30:57.964422685 +0000
@@ -8,7 +8,7 @@
 } || {
 _comp_dpkg_hold_packages()
 {
-    command grep -B 2 'hold' /var/lib/dpkg/status | \
+    command ggrep -B 2 'hold' /var/lib/dpkg/status | \
         awk "/Package: $1/ { print \$2 }"
 }
 }
@@ -65,7 +65,7 @@
             ;;
         -t|--target-release|--default-release)
             COMPREPLY=( $( apt-cache policy | \
-                command grep "release.o=Debian,a=$cur" | \
+                command ggrep "release.o=Debian,a=$cur" | \
                 command sed -e "s/.*a=\(\w*\).*/\1/" | uniq 2> /dev/null ) )
             return
             ;;
diff -Nurb bash-completion-2.5.orig/completions/cfrun bash-completion-2.5/completions/cfrun
--- bash-completion-2.5.orig/completions/cfrun	2017-06-02 18:26:34.795569838 +0000
+++ bash-completion-2.5/completions/cfrun	2017-06-02 18:28:36.820944012 +0000
@@ -33,7 +33,7 @@
                 done
                 [[ ! -f $hostfile ]] && return
 
-                COMPREPLY=( $(compgen -W  "$( command grep -v \
+                COMPREPLY=( $(compgen -W  "$( command ggrep -v \
                     -E '(=|^$|^#)' $hostfile )" -- "$cur" ) )
             fi
             ;;
diff -Nurb bash-completion-2.5.orig/completions/dict bash-completion-2.5/completions/dict
--- bash-completion-2.5.orig/completions/dict	2017-06-02 18:26:34.797167061 +0000
+++ bash-completion-2.5/completions/dict	2017-06-02 18:28:36.786069109 +0000
@@ -56,7 +56,7 @@
         # in a pattern instead.
         if [[ $cur == +([-A-Za-z0-9/.]) ]]; then
             COMPREPLY=( $( compgen -W \
-                '$( command grep "^${cur//./\\.}" $dictfile )' -- "$cur" ) )
+                '$( command ggrep "^${cur//./\\.}" $dictfile )' -- "$cur" ) )
         else
             COMPREPLY=( $( compgen -W '$( cat $dictfile )' -- "$cur" ) )
         fi
diff -Nurb bash-completion-2.5.orig/completions/dpkg bash-completion-2.5/completions/dpkg
--- bash-completion-2.5.orig/completions/dpkg	2017-06-02 18:26:34.790835521 +0000
+++ bash-completion-2.5/completions/dpkg	2017-06-02 18:28:36.836894812 +0000
@@ -8,8 +8,8 @@
 } || {
 _comp_dpkg_installed_packages()
 {
-    command grep -A 1 "Package: $1" /var/lib/dpkg/status 2>/dev/null | \
-        command grep -B 1 -Ee "ok installed|half-installed|unpacked| \
+    command ggrep -A 1 "Package: $1" /var/lib/dpkg/status 2>/dev/null | \
+        command ggrep -B 1 -Ee "ok installed|half-installed|unpacked| \
             half-configured" \
             -Ee "^Essential: yes" | \
         awk "/Package: $1/ { print \$2 }" 2>/dev/null
@@ -24,8 +24,8 @@
 } || {
 _comp_dpkg_purgeable_packages()
 {
-    command grep -A 1 "Package: $1" /var/lib/dpkg/status 2>/dev/null | \
-        command grep -B 1 -Ee "ok installed|half-installed|unpacked| \
+    command ggrep -A 1 "Package: $1" /var/lib/dpkg/status 2>/dev/null | \
+        command ggrep -B 1 -Ee "ok installed|half-installed|unpacked| \
             half-configured|config-files" \
             -Ee "^Essential: yes" | \
         awk "/Package: $1/ { print \$2 }" 2>/dev/null
diff -Nurb bash-completion-2.5.orig/completions/gcc bash-completion-2.5/completions/gcc
--- bash-completion-2.5.orig/completions/gcc	2017-06-02 18:26:34.805024802 +0000
+++ bash-completion-2.5/completions/gcc	2017-06-02 18:28:36.817852884 +0000
@@ -48,16 +48,16 @@
 } &&
 complete -F _gcc gcc g++ gfortran g77 g95 gcj gpc &&
 {
-    cc  --version 2>/dev/null | command grep -q GCC || \
+    cc  --version 2>/dev/null | command ggrep -q GCC || \
         [[ $( _realcommand cc ) == *gcc* ]] && \
         complete -F _gcc cc  || complete -F _minimal cc
-    c++ --version 2>/dev/null | command grep -q GCC || \
+    c++ --version 2>/dev/null | command ggrep -q GCC || \
         [[ $( _realcommand c++ ) == *g++* ]] && \
         complete -F _gcc c++ || complete -F _minimal c++
-    f77 --version 2>/dev/null | command grep -q GCC || \
+    f77 --version 2>/dev/null | command ggrep -q GCC || \
         [[ $( _realcommand f77 ) == *gfortran* ]] && \
         complete -F _gcc f77 || complete -F _minimal f77
-    f95 --version 2>/dev/null | command grep -q GCC || \
+    f95 --version 2>/dev/null | command ggrep -q GCC || \
         [[ $( _realcommand f95 ) == *gfortran* ]] && \
         complete -F _gcc f95 || complete -F _minimal f95
 }
diff -Nurb bash-completion-2.5.orig/completions/gnokii bash-completion-2.5/completions/gnokii
--- bash-completion-2.5.orig/completions/gnokii	2017-06-02 18:26:34.796775773 +0000
+++ bash-completion-2.5/completions/gnokii	2017-06-02 18:28:36.779952479 +0000
@@ -222,7 +222,7 @@
     # safer to use LANG=C
     local all_cmd="$( LANG=C _parse_help $1 "--help all" )"
     # these 2 below are allowed in combination with others
-    local main_cmd=$( command grep -v -- '--config\|--phone' <<<"$all_cmd" )
+    local main_cmd=$( command ggrep -v -- '--config\|--phone' <<<"$all_cmd" )
 
     # don't provide main command completions if one is
     # already on the command line
diff -Nurb bash-completion-2.5.orig/completions/isql bash-completion-2.5/completions/isql
--- bash-completion-2.5.orig/completions/isql	2017-06-02 18:26:34.799555613 +0000
+++ bash-completion-2.5/completions/isql	2017-06-02 18:28:36.772241572 +0000
@@ -7,7 +7,7 @@
     _init_completion || return
 
     [[ -f $ODBCINI ]] \
-        && COMPREPLY=( $( command grep "\[$cur" "$ODBCINI" | tr -d \[\] ) )
+        && COMPREPLY=( $( command ggrep "\[$cur" "$ODBCINI" | tr -d \[\] ) )
 } &&
 complete -F _isql isql
 
diff -Nurb bash-completion-2.5.orig/completions/java bash-completion-2.5/completions/java
--- bash-completion-2.5.orig/completions/java	2017-06-02 18:26:34.796531290 +0000
+++ bash-completion-2.5/completions/java	2017-06-02 18:28:36.807737873 +0000
@@ -63,10 +63,10 @@
         if [[ "$i" == *.@(jar|zip) && -r $i ]]; then
             if type zipinfo &>/dev/null; then
                 COMPREPLY+=( $( zipinfo -1 "$i" "$cur*" 2>/dev/null | \
-                    command grep '^[^$]*\.class$' ) )
+                    command ggrep '^[^$]*\.class$' ) )
             else
                 COMPREPLY+=( $( jar tf "$i" "$cur" | \
-                    command grep '^[^$]*\.class$' ) )
+                    command ggrep '^[^$]*\.class$' ) )
             fi
 
         elif [[ -d $i ]]; then
@@ -107,7 +107,7 @@
         fi
     done
     # keep only packages
-    COMPREPLY=( $( tr " " "\n" <<<"${COMPREPLY[@]}" | command grep "/$" ) )
+    COMPREPLY=( $( tr " " "\n" <<<"${COMPREPLY[@]}" | command ggrep "/$" ) )
     # remove packages extension
     COMPREPLY=( ${COMPREPLY[@]%/} )
     # convert path syntax to package syntax
diff -Nurb bash-completion-2.5.orig/completions/lintian bash-completion-2.5/completions/lintian
--- bash-completion-2.5.orig/completions/lintian	2017-06-02 18:26:34.799603817 +0000
+++ bash-completion-2.5/completions/lintian	2017-06-02 18:28:36.811465452 +0000
@@ -8,7 +8,7 @@
     if [[ "$cur" == *, ]]; then
         search=${cur//,/ }
         for item in $search; do
-            match=$( command grep -nE "^Tag: $item$" \
+            match=$( command ggrep -nE "^Tag: $item$" \
                 /usr/share/lintian/checks/*.desc | cut -d: -f1 )
             tags=$( command sed -e "s/\<$item\>//g" <<<$tags )
         done
@@ -29,7 +29,7 @@
     if [[ "$cur" == *, ]]; then
         search=${cur//,/ }
         for item in $search; do
-            match=$( command grep -nE "^(Check-Script|Abbrev): $item$" \
+            match=$( command ggrep -nE "^(Check-Script|Abbrev): $item$" \
                 /usr/share/lintian/checks/*.desc | cut -d: -f1 )
             todisable=$( awk '/^(Check-Script|Abbrev)/ { print $2 }' $match )
             for name in $todisable; do
@@ -53,7 +53,7 @@
     if [[ "$cur" == *, ]]; then
         search=${cur//,/ }
         for item in $search; do
-            match=$( command grep -nE "^Collector: $item$" \
+            match=$( command ggrep -nE "^Collector: $item$" \
                 /usr/share/lintian/collection/*.desc | cut -d: -f1 )
             infos=$( command sed -e "s/\<$item\>//g" <<<$infos )
         done
diff -Nurb bash-completion-2.5.orig/completions/minicom bash-completion-2.5/completions/minicom
--- bash-completion-2.5.orig/completions/minicom	2017-06-02 18:26:34.792830608 +0000
+++ bash-completion-2.5/completions/minicom	2017-06-02 18:28:36.795298318 +0000
@@ -32,7 +32,7 @@
     COMPREPLY=(
         $( printf '%s\n' /etc/minirc.* /etc/minicom/minirc.* ~/.minirc.* \
             | command sed -e '/\*$/d' -e 's/^.*minirc\.//' \
-            | command grep "^${cur}" ) )
+            | command ggrep "^${cur}" ) )
 } &&
 complete -F _minicom -o default minicom
 
diff -Nurb bash-completion-2.5.orig/completions/pkgadd bash-completion-2.5/completions/pkgadd
--- bash-completion-2.5.orig/completions/pkgadd	2017-06-02 18:26:34.802646981 +0000
+++ bash-completion-2.5/completions/pkgadd	2017-06-02 18:28:36.775861870 +0000
@@ -49,7 +49,7 @@
                         pkginst_list="${pkginst_list[@]}"
                 else
                     pkginst_list=$(strings $(dequote $device) | \
-                        command grep "^PKG=" | sort -u | cut -d= -f2)
+                        command ggrep "^PKG=" | sort -u | cut -d= -f2)
             fi
             COMPREPLY=( $(compgen -W "$pkginst_list" -- ${cur}) )
         fi
diff -Nurb bash-completion-2.5.orig/completions/portinstall bash-completion-2.5/completions/portinstall
--- bash-completion-2.5.orig/completions/portinstall	2017-06-02 18:26:34.794206553 +0000
+++ bash-completion-2.5/completions/portinstall	2017-06-02 18:28:36.782902179 +0000
@@ -18,9 +18,9 @@
 
     [[ "$prev" == -l || "$prev" == -L || "$prev" == -o ]] && return
 
-    COMPREPLY=( $( command grep -E "^$cur" 2>/dev/null < $indexfile | \
+    COMPREPLY=( $( command ggrep -E "^$cur" 2>/dev/null < $indexfile | \
         cut -d'|' -f1 ) )
-    COMPREPLY2=( $( command grep -E "^[^\|]+\|$portsdir$cur" 2>/dev/null \
+    COMPREPLY2=( $( command ggrep -E "^[^\|]+\|$portsdir$cur" 2>/dev/null \
         < $indexfile | cut -d'|' -f2 ) )
     COMPREPLY2=( ${COMPREPLY2[@]#$portsdir} )
     COMPREPLY+=( "${COMPREPLY2[@]}" )
diff -Nurb bash-completion-2.5.orig/completions/povray bash-completion-2.5/completions/povray
--- bash-completion-2.5.orig/completions/povray	2017-06-02 18:26:34.807832506 +0000
+++ bash-completion-2.5/completions/povray	2017-06-02 18:28:36.827157420 +0000
@@ -21,14 +21,14 @@
             ;;
         [-+]O*)
             # guess what output file type user may want
-            case $( ( IFS=$'\n'; command grep '^[-+]F' <<<"${words[*]}" ) ) in
+            case $( ( IFS=$'\n'; command ggrep '^[-+]F' <<<"${words[*]}" ) ) in
                 [-+]FN) oext=png ;;
                 [-+]FP) oext=ppm ;;
                 [-+]F[CT]) oext=tga ;;
                 *) oext=$defoext ;;
             esac
             # complete filename corresponding to previously specified +I
-            COMPREPLY=( $( ( IFS=$'\n'; command grep '^[-+]I' <<<"${words[*]}" ) ) )
+            COMPREPLY=( $( ( IFS=$'\n'; command ggrep '^[-+]I' <<<"${words[*]}" ) ) )
             COMPREPLY=( ${COMPREPLY[@]#[-+]I} )
             COMPREPLY=( ${COMPREPLY[@]/%.pov/.$oext} )
             cur="${povcur#[-+]O}" # to confuse _filedir
diff -Nurb bash-completion-2.5.orig/completions/rdesktop bash-completion-2.5/completions/rdesktop
--- bash-completion-2.5.orig/completions/rdesktop	2017-06-02 18:26:34.807588042 +0000
+++ bash-completion-2.5/completions/rdesktop	2017-06-02 18:28:36.789143415 +0000
@@ -9,7 +9,7 @@
         -k)
             COMPREPLY=( $( command ls \
                 /usr/share/rdesktop/keymaps 2>/dev/null | \
-                command grep -E -v '(common|modifiers)' ) )
+                command ggrep -E -v '(common|modifiers)' ) )
             COMPREPLY+=( $( command ls $HOME/.rdesktop/keymaps 2>/dev/null ) )
             COMPREPLY+=( $( command ls ./keymaps 2>/dev/null ) )
             COMPREPLY=( $( compgen -W '${COMPREPLY[@]}' -- "$cur" ) )
diff -Nurb bash-completion-2.5.orig/completions/reportbug bash-completion-2.5/completions/reportbug
--- bash-completion-2.5.orig/completions/reportbug	2017-06-02 18:26:34.796240978 +0000
+++ bash-completion-2.5/completions/reportbug	2017-06-02 18:28:36.792350493 +0000
@@ -48,7 +48,7 @@
             ;;
         --from-buildd)
             COMPREPLY=( $( compgen -S "_" -W '$( apt-cache dumpavail | \
-                command grep "^Source: $cur" | sort -u | cut -f2 -d" " )' ))
+                command ggrep "^Source: $cur" | sort -u | cut -f2 -d" " )' ))
             return
             ;;
         *)
diff -Nurb bash-completion-2.5.orig/completions/rpm bash-completion-2.5/completions/rpm
--- bash-completion-2.5.orig/completions/rpm	2017-06-02 18:26:34.790686918 +0000
+++ bash-completion-2.5/completions/rpm	2017-06-02 18:28:36.799818216 +0000
@@ -106,7 +106,7 @@
                 esac
                 COMPREPLY=( $( compgen -W "$( $1 -qa --nodigest --nosignature \
                     --queryformat=\"%{$fmt}\\n\" 2>/dev/null |
-                    command grep -vF '(none)' )" -- "$cur" ) )
+                    command ggrep -vF '(none)' )" -- "$cur" ) )
             fi
             return
             ;;
diff -Nurb bash-completion-2.5.orig/completions/slackpkg bash-completion-2.5/completions/slackpkg
--- bash-completion-2.5.orig/completions/slackpkg	2017-06-02 18:26:34.808640194 +0000
+++ bash-completion-2.5/completions/slackpkg	2017-06-02 18:28:36.840223135 +0000
@@ -79,12 +79,12 @@
             COMPREPLY+=( $( compgen -W 'a ap d e f k kde kdei l n t tcl x
                 xap xfce y' -- "$cur" ) )
             COMPREPLY+=( $( cut -f 6 -d\  "${WORKDIR}/pkglist" 2> /dev/null | \
-                command grep "^$cur" ) )
+                command ggrep "^$cur" ) )
             return
             ;;
         info)
             COMPREPLY=( $( cut -f 6 -d\  "${WORKDIR}/pkglist" 2> /dev/null | \
-                command grep "^$cur" ) )
+                command ggrep "^$cur" ) )
             return
             ;;
         update)
diff -Nurb bash-completion-2.5.orig/completions/wvdial bash-completion-2.5/completions/wvdial
--- bash-completion-2.5.orig/completions/wvdial	2017-06-02 18:26:34.793920729 +0000
+++ bash-completion-2.5/completions/wvdial	2017-06-02 18:28:36.843191754 +0000
@@ -34,7 +34,7 @@
             # parse config files for sections and
             # remove default section
             COMPREPLY=( $( command sed -ne "s|^\[Dialer \($cur.*\)\]$|\1|p" $config \
-                2>/dev/null | command grep -v '^Defaults$'))
+                2>/dev/null | command ggrep -v '^Defaults$'))
             # escape spaces
             COMPREPLY=${COMPREPLY// /\\ }
             ;;
diff -Nurb bash-completion-2.5.orig/test/lib/completions/dpkg.exp bash-completion-2.5/test/lib/completions/dpkg.exp
--- bash-completion-2.5.orig/test/lib/completions/dpkg.exp	2017-06-02 18:26:34.716655547 +0000
+++ bash-completion-2.5/test/lib/completions/dpkg.exp	2017-06-02 18:28:36.750135366 +0000
@@ -16,7 +16,7 @@
 
 
 # Build list of installed packages
-if {[assert_exec {dpkg --get-selections | command grep \[\[:space:\]\]install$ | cut -f1} packages "" "untested"]} {
+if {[assert_exec {dpkg --get-selections | command ggrep \[\[:space:\]\]install$ | cut -f1} packages "" "untested"]} {
     assert_complete $packages "dpkg -L "
 }
 sync_after_int
diff -Nurb bash-completion-2.5.orig/test/lib/completions/feh.exp bash-completion-2.5/test/lib/completions/feh.exp
--- bash-completion-2.5.orig/test/lib/completions/feh.exp	2017-06-02 18:26:34.735927627 +0000
+++ bash-completion-2.5/test/lib/completions/feh.exp	2017-06-02 18:29:28.777399390 +0000
@@ -17,7 +17,7 @@
 sync_after_int
 
 
-if {[assert_exec {feh --help | grep "man feh" || exit 1} "" "" "unsupported"]} {
+if {[assert_exec {feh --help | command ggrep "man feh" || exit 1} "" "" "unsupported"]} {
     set test "--lis<TAB> should complete \"--list\""
     assert_complete "--list" "feh --lis" $test
 }
diff -Nurb bash-completion-2.5.orig/test/lib/completions/kldunload.exp bash-completion-2.5/test/lib/completions/kldunload.exp
--- bash-completion-2.5.orig/test/lib/completions/kldunload.exp	2017-06-02 18:26:34.740080423 +0000
+++ bash-completion-2.5/test/lib/completions/kldunload.exp	2017-06-02 18:29:57.371695777 +0000
@@ -11,7 +11,7 @@
 setup
 
 
-if {[catch {eval exec {kldstat | grep "\.ko$"}}] == 0} {
+if {[catch {eval exec {kldstat | command ggrep "\.ko$"}}] == 0} {
     assert_complete_any "kldunload "
 } else {
     assert_no_complete "kldunload "
diff -Nurb bash-completion-2.5.orig/test/lib/completions/look.exp bash-completion-2.5/test/lib/completions/look.exp
--- bash-completion-2.5.orig/test/lib/completions/look.exp	2017-06-02 18:26:34.724997673 +0000
+++ bash-completion-2.5/test/lib/completions/look.exp	2017-06-02 18:30:12.539774049 +0000
@@ -11,7 +11,7 @@
 setup
 
 
-if {[assert_exec {look foo | grep ^foo} words {} "unsupported"]} {
+if {[assert_exec {look foo | command ggrep ^foo} words {} "unsupported"]} {
     assert_complete $words "look foo"
 }
 
diff -Nurb bash-completion-2.5.orig/test/lib/completions/sysctl.exp bash-completion-2.5/test/lib/completions/sysctl.exp
--- bash-completion-2.5.orig/test/lib/completions/sysctl.exp	2017-06-02 18:26:34.718931440 +0000
+++ bash-completion-2.5/test/lib/completions/sysctl.exp	2017-06-02 18:29:40.059564305 +0000
@@ -11,7 +11,7 @@
 setup
 
 
-if {[assert_exec {sysctl -N -a 2>/dev/null | grep ^kern | sort -u} values]} {
+if {[assert_exec {sysctl -N -a 2>/dev/null | command ggrep ^kern | sort -u} values]} {
     set test "sysctl kern should show completions"
     assert_complete $values "sysctl kern" $test -expect-cmd-minus kern
 }
diff -Nurb bash-completion-2.5.orig/test/lib/library.exp bash-completion-2.5/test/lib/library.exp
--- bash-completion-2.5.orig/test/lib/library.exp	2017-06-02 18:26:34.710844011 +0000
+++ bash-completion-2.5/test/lib/library.exp	2017-06-02 18:28:36.767530729 +0000
@@ -670,7 +670,7 @@
         # Retrieving hosts is successful?
     if { [catch {exec bash -c {
         type avahi-browse >&/dev/null \
-            && avahi-browse -cpr _workstation._tcp 2>/dev/null | command grep ^= | cut -d\; -f7 | sort -u
+            && avahi-browse -cpr _workstation._tcp 2>/dev/null | command ggrep ^= | cut -d\; -f7 | sort -u
     }} hosts] } {
         # No, retrieving hosts yields error;
             # Reset hosts
